pkg/exchange/okex/exchange.go

```diff
@@ -339,3 +339,30 @@ func (e *Exchange) QueryKLines(ctx context.Context, symbol string, interval type
 	return klines, nil
 
 }
+
+func (e *Exchange) QueryOrder(ctx context.Context, q types.OrderQuery) (*types.Order, error) {
+	if len(q.Symbol) == 0 {
+		return nil, errors.New("okex.QueryOrder: InstrumentID is required parameter")
+	}
+	if len(q.OrderID) == 0 && len(q.ClientOrderID) == 0 {
+		return nil, errors.New("okex.QueryOrder: ordId or clOrdId is required parameter")
+	}
+	req := e.client.TradeService.NewGetOrderDetailsRequest()
+	req.InstrumentID(q.Symbol).
+		OrderID(q.OrderID).
+		ClientOrderID(q.ClientOrderID)
+
+	var order *okexapi.OrderDetails
+	order, err := req.Do(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	isMargin := false
+	if order.InstrumentType == string(okexapi.InstrumentTypeMARGIN) {
+		isMargin = true
+	}
```

Comment:
The conversion does not depend on any variables within `Exchange` struct. Is there any reason not to check the `order.InstrumentType` in the `toGlobalOrder` function?

Reply:
Updated. I check `order.InstrumentType` in the `toGlobalOrder` function now